lib/passport-wsfed-saml2/utils.js

```diff
@@ -85,3 +85,24 @@ exports.getEncoding = function(xml){
     return;
   }
 }
+
+/**
+ * Safely compare two string. Type validation and length comparison are inspired in the
+ * cryptiles.fixedTimeComparison method and kept to avoid the linear validation when
+ * comparing the two strings at the end of the method.
+ *
+ * @param a
+ * @param b
+ * @return {boolean}
+ */
+exports.stringCompare = function(a,b) {
```

Comment:
this is not fixed time, why dont we use https://nodejs.org/api/crypto.html#crypto_crypto_timingsafeequal_a_b

Reply:
If you look at the previous code is doing this. I think that this is safer than convert this string to a buffer and check using that primitive.

Reply:
I think in this case is fine, but if a fixed time comparison is needed this is not enough (I checked the spec and does not mention anything in this case)